File: benchmarks/struct_records_reuse/struct_records_reuse.go
package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func makePerson(i int) Person {
	return Person{Name: fmt.Sprintf("user%d", i), Age: 20 + i%50}
}

func birthday(p Person) Person {
	return Person{Name: p.Name, Age: p.Age + 1}
}

func score(p Person) int {
	return len(p.Name) + p.Age
}

func main() {
	total := 0
	for i := 0; i < 500000; i++ {
		p := makePerson(i)
		q := birthday(p)
		total += score(p) + score(q)
	}
	if total != 54777780 {
		panic("struct_records_reuse checksum")
	}
	fmt.Println("assert passed, struct_records_reuse is correct")
}

File: benchmarks/struct_records_reuse/struct_records_reuse.swift
struct Person {
    var name: String
    var age: Int
}

func makePerson(_ i: Int) -> Person {
    Person(name: "user" + String(i), age: 20 + i % 50)
}

func birthday(_ p: Person) -> Person {
    Person(name: p.name, age: p.age + 1)
}

func score(_ p: Person) -> Int {
    p.name.utf8.count + p.age
}

var total = 0
for i in 0..<500000 {
    let p = makePerson(i)
    let q = birthday(p)
    total += score(p) + score(q)
}
precondition(total == 54777780, "struct_records_reuse checksum")
print("assert passed, struct_records_reuse is correct")
